subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -267,8 +267,8 @@ static void tls_debug(void *ctx, int level, const char *file,
 		}
 	}
 
-	NET_DBG("%s:%04d: |%d| %s", basename, line, level,
-		str);
+	/* Remove extra EOL to improve log readability. */
+	NET_DBG("%s:%04d: |%d| %.*s", basename, line, level, strlen(str) - 1, str);
```

Comment:
We should drop this change from this PR, in favour of https://github.com/zephyrproject-rtos/zephyr/pull/46572